port/unix/omrsysinfo.c

```diff
@@ -4497,21 +4497,21 @@ omrsysinfo_cgroup_subsystem_iterator_next(struct OMRPortLibrary *portLibrary, st
 		if ((SINGLE_CGROUP_METRIC == subsystemMetricMapElement->metricElementsCount) && (NULL != currentElement->metricKeyInFile)) {
 			isSingleLineKeyValue = TRUE;
 		}
-		rc = readCgroupMetricFromFile(portLibrary, state->subsystemid, subsystemMetricMapElement->metricFileName, currentElement->metricKeyInFile, metricElement->value, &(*state).fileContent, isSingleLineKeyValue, sizeRef);
+		rc = readCgroupMetricFromFile(portLibrary, state->subsystemid, subsystemMetricMapElement->metricFileName, currentElement->metricKeyInFile, metricElement->value, &(state->fileContent), isSingleLineKeyValue, sizeRef);
 		if (0 == rc) {
 			if(NULL == state->fileContent) {
 				state->multiLineCounter = SINGLE_CGROUP_METRIC;
 			} else {
 				state->multiLineCounter = 0;
 			}
 		} else {
+			state->count = state->count + 1;
```

Comment:
@DanHeidinga  Thanks for the review.

> `rc` will be an error code in this case. Won't that terminate correctly coded uses of the iterator?

In my implementation I'm using these api and printing the metrics based on `rc` value but not terminating the iterator. Iterator termination is completely based on `omrsysinfo_cgroup_subsystem_iterator_hasNext` api which is internally dependent on the check for `state->count`. So if the `file is not found` and `state->count` is unchanged its ending up in searching for same file again and again (A bug in the way i implemented these api's).

> If the intention is to allow the iterator to continue, and I'm not clear on why it should

I thought it would be good if iterator continues irrespective of `rc` value as there might be errors in accessing the metric from file such as `file not found` or `metric not found` in file, these shouldn't interrupt the iterator from reading other metrics in the list which might be available. Please correct me if I'm wrong 

> then the rc will need to be changed.

Changing the `rc` would be a good option as it will alert the caller that the current metric is not available. 

> Also, use `state->count += 1`

Sure will be using it.